tests/spicy/types/vector/parse-while-foreach.spicy

```diff
@@ -0,0 +1,23 @@
+# @TEST-EXEC: printf 12345A12345B12345C | spicy-driver -d %INPUT >output
+# @TEST-EXEC: btest-diff output
+#
+# @TEST-DOC: Check that foreach-hooks execute before a &while condition is evaluated.
+
+module Test;
+
+type subtype = unit {
+    x: bytes &size=5;
+    y: int8;
+};
+
+public type test = unit {
+    element: subtype()[] &while=self.continue_parsing foreach {
+        if ( $$.y == 'C' ) {
+            self.continue_parsing = False ;
+        }
+    }
```

Comment:
TBH, I find the behavior of this confusing and I would have expected `foreach` to operate on the values ending up in the unit, not everything seen. Since `&while` is an attribute which will always appear before a `foreach` and its statement/block, reading the field declaration seems to suggest that we'd examine `continue_parsing` before including the element. Do you think it would make sense change the behavior to have that order instead?

Reply:
I wouldn't change the parsing order (because that would be problematic for other parts), but yeah, I go back and forth on whether this is a good change, too. Your comment makes me feel inclined to change it back. @0xxon  thoughts?

Reply:
Changing it back after all.

Reply:
I actually think I expressed myself wrong in the bug report - sorry. I actually want exactly the behavior that @bbannier sketches.

Thinking about it again, the buggy behavior that I wanted to describe (...but did not actually describe) is that the following happend:

&while executed, seeing continue_parsing=True, adding the element.
foreach execute, setting continue_parsing to False, because this was to be the lase element.
...and after that, another parsing step was started, and another element was added, even though continue_parsing was already set to false.

This is also why I said that the behavior typically is correct and that I found an edge-case where it does not seem to work correctly. I think I can reliably trigger this - but it is part of a larger example.